src/arch/armv8/aborts.c

```diff
@@ -133,15 +139,17 @@ void sysreg_handler(unsigned long iss, unsigned long far, unsigned long il, unsi
     }
 
     emul_handler_t handler = vm_emul_get_reg(cpu()->vcpu->vm, reg_addr);
-    if(handler != NULL){
+    if (handler != NULL) {
         struct emul_access emul;
         emul.addr = reg_addr;
         emul.width = 8;
         emul.write = iss & ESR_ISS_SYSREG_DIR ? false : true;
-        emul.reg = bit64_extract(iss, ESR_ISS_SYSREG_REG_OFF, ESR_ISS_SYSREG_REG_LEN);
-        emul.reg_high = bit64_extract(iss, ESR_ISS_SYSREG_REG2_OFF, ESR_ISS_SYSREG_REG2_LEN);
+        emul.reg =
+            bit64_extract(iss, ESR_ISS_SYSREG_REG_OFF, ESR_ISS_SYSREG_REG_LEN);
+        emul.reg_high = bit64_extract(iss, ESR_ISS_SYSREG_REG2_OFF,
+            ESR_ISS_SYSREG_REG2_LEN);
```

Comment:
In cases like this personally I'd prefer to have AlignAfterOpenBracket=BAS_Align

Reply:
My opinion is that this becomes a bit weird in some corner cases, as not aligning is pretty much linear. See an example of what I'm talking about below:

```c
void as_init(struct addr_space* as, enum AS_TYPE type, asid_t id, pte_t* root_pt, colormap_t colors)
{
    as->type = type;
    as->pt.dscr = type == AS_HYP || type == AS_HYP_CPY ? hyp_pt_dscr : vm_pt_dscr;
    as->colors = colors;
    as->lock = SPINLOCK_INITVAL;
    as->id = id;

    if (root_pt == NULL) {
        size_t n = NUM_PAGES(pt_size(&as->pt, 0));
        root_pt = (pte_t*)mem_alloc_page(n,
                                         type == AS_HYP || type == AS_HYP_CPY ? SEC_HYP_PRIVATE :
                                                                                SEC_HYP_VM,
                                         true);
        memset((void*)root_pt, 0, n * PAGE_SIZE);
    }
    as->pt.root = root_pt;

    as_arch_init(as);
}
```

Reply:
Code could be written to avoid these cases, but nevertheless..

Reply:
IMO, i would prefer to defer that option since it leads to strange formats like @josecm as showcase..

Reply:
IMO this case in particular would be better written by separating the type identification from the function invocation into different lines and I don't see it as reason to not use BAS_Align. IMO formatting should also discourage writing code that is harder to read.

Reply:
Yes, I totally get your point and agree. But we then need to manually scan the current code and make this refactoring so we don't get other cases like this.

Reply:
For the time being, I would for leaving the code as it is for now

Reply:
I made the changes here
https://github.com/bao-project/bao-hypervisor/tree/tmp/style_clang_format_align

there are two commits, one where I think align works well, the other where I think it's not ideal.
IMO the benefits outweigh the costs, but I'm fine with whatever decision for the time being

Reply:
When enabling that option, I can't find a way to get around of it putting multiple entries per line on arrays, instead on one entry per line. Like [here](https://github.com/bao-project/bao-hypervisor/blob/a0adc4adeedc7e9ba0ed83768fd06fec090f9399/src/arch/armv8/aborts.c#L154C1-L159C3). I'd suggest we keep it as it is right now.

Reply:
We'll leave this as is for now.